### ISSUE TITLE
Deprecation warnings when using patch_sklearn()

### ISSUE THREAD

**Issue by DavidCohen2** [uxlfoundation/scikit-learn-intelex#2321]
---
name: "Bug_report"
about: Create a report to help us improve
title: 'Deprecation warnings when using patch_sklearn()'
labels: bug
assignees: ''

---

**Describe the bug**  
When enabling Intel optimizations via `patch_sklearn()` from scikit-learn-intelex, several `FutureWarning` messages are printed indicating that `'force_all_finite' was renamed to 'ensure_all_finite' in 1.6 and will be removed in 1.8`. These warnings do not appear when the patch is commented out, suggesting an unintended side effect of the extension.

**To Reproduce**  
1. Install the following packages:
   - scikit-learn 1.6.1
   - scikit-learn-intelex 2025.1.0
2. Create a Python script with the following code (indented for clarity):
    
    import platform
    import sys
    from sklearnex import patch_sklearn
    patch_sklearn()  # Enable Intel optimizations for scikit-learn 

    from sklearn.svm import SVC
    from sklearn.datasets import make_classification

    # Create data and train an SVC model
    X, y = make_classification(n_samples=100, n_features=10, random_state=42)
    clf = SVC()
    clf.fit(X, y)

    # Print system details
    print("### System Info ###")
    print(f"Python Version: {sys.version}")
    print(f"Platform: {platform.platform()}")
    print(f"Processor: {platform.processor()}")

3. Run the script.
4. Observe that the output includes the Intel extension banner and multiple `FutureWarning` messages regarding the `force_all_finite` parameter.
5. Comment out the `patch_sklearn()` call and re-run the script to see that the warnings are not present.

**Expected behavior**  
Enabling scikit-learn-intelex via `patch_sklearn()` should not trigger deprecation warnings from scikit-learn. The extension should seamlessly integrate Intel optimizations without surfacing warnings related to parameter renaming.

**Output/Screenshots**  
With `patch_sklearn()` enabled:

Intel(R) Extension for Scikit-learn* enabled (https://github.com/intel/scikit-learn-intelex)
c:\Users\david\anaconda3\envs\mfs_1\Lib\site-packages\sklearn\utils\deprecation.py:151: FutureWarning: 'force_all_finite' was renamed to 'ensure_all_finite' in 1.6 and will be removed in 1.8.
  warnings.warn(
... (similar warnings repeated)
### System Info ###
Python Version: 3.13.2 | packaged by conda-forge | (main, Feb 17 2025, 13:52:56) [MSC v.1942 64 bit (AMD64)]
Platform: Windows-11-10.0.22000-SP0
Processor: Intel64 Family 6 Model 151 Stepping 2, GenuineIntel

With `patch_sklearn()` commented out:
No warnings are produced, only the system information is printed.

**Environment:**  
- OS: Windows 11 (Version: Windows-11-10.0.22000-SP0)
- Python: 3.13.2 (packaged by conda-forge)
- scikit-learn: 1.6.1
- scikit-learn-intelex: 2025.1.0
- Processor: Intel64 Family 6 Model 151 Stepping 2, GenuineIntel

**Comment by david-cortes-intel** [comment]
Thanks for reporting the problem. I can confirm that it does indeed issue this warning when applying the patch in an environment with the version numbers you provided.

This however has been fixed in the latest development version of `scikit-learn-intelex`, so the fix should be available in the next version release.

**Comment by Alexsandruss** [comment]
Fixed in https://github.com/uxlfoundation/scikit-learn-intelex/pull/2346
